libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -590,6 +590,26 @@ void AP_GPS::send_blob_start(uint8_t instance)
     }
 #endif // AP_GPS_NMEA_ENABLED
 
+    // the following devices don't have init blobs:
+    switch (_type[instance]) {
+#if AP_GPS_SBF_ENABLED
+    case GPS_TYPE_SBF:
+#endif //AP_GPS_SBF_ENABLED
+#if AP_GPS_GSOF_ENABLED
+    case GPS_TYPE_GSOF:
+#endif //AP_GPS_GSOF_ENABLED
+#if AP_GPS_NOVA_ENABLED
+    case GPS_TYPE_NOVA:
+#endif //AP_GPS_NOVA_ENABLED
+#if HAL_SIM_GPS_ENABLED
+    case GPS_TYPE_SITL:
+#endif  // HAL_SIM_GPS_ENABLED
+        send_blob_start(instance, nullptr, 0);
```

Comment:
Not a blocker but what's the value of doing this at all? Why not just do the return here?

It's also a slightly larger diff, but I'd prefer to see the instances above be moved into the same switch here. For example `GPS_TYPE_HEMI` fits easily, the ublox ones would need an if clause with early return still, but it would be nice to be able to actually read through the full switch instead of both early special cases and then a switch. This isn't a blocker though.

Reply:
> Not a blocker but what's the value of doing this at all? Why not just do the return here?

Rather than calling send_blob_start and then returning?  I thought it a good idea to make sure that we weren't sending any blob - no reason to think here that the init blob currently in there is relevant to the GPS we're currently trying.  We don't zero the init-blob-sending-state when we try a new type.

> It's also a slightly larger diff, but I'd prefer to see the instances above be moved into the same switch here. For example `GPS_TYPE_HEMI` fits easily, the ublox ones would need an if clause with early return still, but it would be nice to be able to actually read through the full switch instead of both early special cases and then a switch. This isn't a blocker though.

Yeah, I resisted doing that for this PR to keep the patchset small.  I did spend a few minutes staring at the uBlox stuff to see how it would fit into a switch and came away with the impression of "kinda-sorta, but a chunk of ugly diff" - no impossible but not a good idea for this PR....

Reply:
> Rather than calling send_blob_start and then returning? I thought it a good idea to make sure that we weren't sending any blob - no reason to think here that the init blob currently in there is relevant to the GPS we're currently trying. We don't zero the init-blob-sending-state when we try a new type.

So on the one hand it's true. On the other hand we could just return and in the general case we never would have sent anything so there would be no blob pending. (IE think of a system that booted as one of these GPS units). The only time anythings pending is if you change type at runtime, but if you are doing this then you've already been sending the blob out before this to the GPS anyways, so in practice it is a bit of a moot point.

> Yeah, I resisted doing that for this PR to keep the patchset small. I did spend a few minutes staring at the uBlox stuff to see how it would fit into a switch and came away with the impression of "kinda-sorta, but a chunk of ugly diff" - no impossible but not a good idea for this PR....

I think it's worth doing, but doing it as a follow up makes a lot of sense.